main.go

```diff
@@ -114,12 +127,24 @@ func main() {
 	// returns nil if not provided and falls back to simple TCP.
 	tlsConfig := createTLSConfig(*esCA, *esClientCert, *esClientPrivateKey, *esInsecureSkipVerify)
 
+	httpTransport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+		Proxy:           http.ProxyFromEnvironment,
+	}
+
 	httpClient := &http.Client{
-		Timeout: *esTimeout,
-		Transport: &http.Transport{
-			TLSClientConfig: tlsConfig,
-			Proxy:           http.ProxyFromEnvironment,
-		},
+		Timeout:   *esTimeout,
+		Transport: httpTransport,
+	}
+
+	if *esApiKey != "" {
```

Comment:
I think it would make sense to move this if statement above the httpClient variable above. That would allow only overwriting the httpTransport variable inside the if statement. That would mean there is only one place where the httpClient variable is set.

Reply:
I agree and I tried to only change the `Transport`, but the issue is that both transports don't have the same type. So I couldn't figure it out how to use one variable and assign either one of the transports based on the `esApiKey` variable.
Do you know how to do this?

Reply:
You can define the var ahead of time as the interface type (i.e. `var transport http.RoundTripper`)

Reply:
Ah of course, I needed to use the interface. It's changed, thanks.